python/nx-cugraph/nx_cugraph/classes/graph.py

```diff
@@ -522,11 +581,36 @@ def _get_plc_graph(
             # Mask is all True; don't need anymore
             del self.edge_masks[edge_attr]
             edge_array = self.edge_values[edge_attr]
+        if edge_array is not None:
+            if edge_dtype is not None:
+                edge_dtype = np.dtype(edge_dtype)
+                if edge_array.dtype != edge_dtype:
+                    edge_array = edge_array.astype(edge_dtype)
+            # PLC doesn't handle int edge weights right now, so cast int to float
```

Comment:
No direct tests yet, but iirc this might get exercised by networkx tests. Wouldn't it be nice to have coverage reports that could answer easily this question 😉 ?

In general, we are largely blazing a path ahead, implementing fast, and trying to write good code, and tests are slow to catch up. You're welcome to write some if you'd like ;) . I expect proactive testing will catch up more next year, since maybe we rely on networkx tests too heavily. For now, "best effort" + networkx tests + add regression tests for regressions is my strategy. Convenient coverage reports would be handy to reveal completely untested code.